### PR TITLE
Fix memory leak and parallel process division

### PR DESCRIPTION
**Pull request by thalassemia**
Parallel processes were not properly saving certain class attributes, including `_parallel`, causing them to be lost upon pickling during division.

Additionally, we were not properly ending parallel processes when deleting stores, causing a memory leak. 

Both of these issues are fixed in this PR.

-----------------------------------------------------------------------

By creating this pull request, I agree to the Contributor License
Agreement, which is available in `CLA.md` at the top level of this
repository.

### CODE REVIEW

File: vivarium/core/store.py [hunk 1]
```diff
@@ -1064,13 +1073,14 @@ def _delete_path(self, path):
         if not path:
             self.inner = {}
             self.value = None
-            return self
+            return None
```
Comment (eagmon) [inline vivarium/core/store.py:1076]: is this change required because `self` is deleted?

File: vivarium/core/store.py [hunk 2]
```diff
@@ -1064,13 +1073,14 @@ def _delete_path(self, path):
...
         target = self.get_path(path[:-1])
         remove = path[-1]
         if remove in target.inner:
-            lost = target.inner[remove]
+            # End any parallel processes to be deleted
+            self.recursive_end_process(target.inner[remove])
             del target.inner[remove]
-            return lost
+            return None
```
Comment (eagmon) [inline vivarium/core/store.py:1083]: likewise here, is `lost` deleted? If it is not, then we should try to keep the prior API

File: vivarium/core/process.py [hunk 3]
```diff
@@ -155,16 +155,16 @@ def __init__(self, parameters: Optional[dict] = None) -> None:
...
+        self._parallel = self._parameters.get('_parallel', False)
         self._condition_path: Optional[HierarchyPath] = None
         self._command_result: Any = None
         self._pending_command: Optional[
             Tuple[str, Optional[tuple], Optional[dict]]] = None
 
         # set up the conditional state if a condition key is provided
         if '_condition' in self._parameters:
-            self._condition_path = self._parameters.pop('_condition')
+            self._condition_path = self._parameters.get('_condition')
```
Comment (eagmon) [inline vivarium/core/process.py:167]: why the change from `pop` to `get`?

**Review by eagmon — APPROVED** [review]
Just a few questions before you merge.

File: vivarium/core/store.py (hunk 1 above)
Comment (thalassemia) [reply vivarium/core/store.py:1076]: Nah I just thought it wasn't really necessary. We don't make use of the return value anywhere and this is a private function.

File: vivarium/core/store.py (hunk 2 above)
Comment (thalassemia) [reply vivarium/core/store.py:1083]: Same as above.

File: vivarium/core/store.py (hunk 1 above)
Comment (eagmon) [reply vivarium/core/store.py:1076]: Can you bring it back? It might be part of the store api developed with @WilliamIX. Then we're good to merge

**Comment by eagmon** [comment]
Fantastic @thalassemia!
